pkg/cmd/server/kubernetes/master/controller/volumes.go

```diff
@@ -1,128 +1,29 @@
-func probeRecyclableVolumePlugins(config componentconfig.VolumeConfiguration, recyclerImageName, recyclerServiceAccountName string) ([]volume.VolumePlugin, error) {
-	uid := int64(0)
-	defaultScrubPod := volume.NewPersistentVolumeRecyclerPodTemplate()
-	// TODO: Move the recycler pods to dedicated namespace instead of polluting
-	// openshift-infra.
-	defaultScrubPod.Namespace = "openshift-infra"
-	defaultScrubPod.Spec.ServiceAccountName = recyclerServiceAccountName
-	defaultScrubPod.Spec.Containers[0].Image = recyclerImageName
-	defaultScrubPod.Spec.Containers[0].Command = []string{"/usr/bin/openshift-recycle"}
-	defaultScrubPod.Spec.Containers[0].Args = []string{"/scrub"}
-	defaultScrubPod.Spec.Containers[0].SecurityContext = &kapiv1.SecurityContext{RunAsUser: &uid}
-	defaultScrubPod.Spec.Containers[0].ImagePullPolicy = kapiv1.PullIfNotPresent
-
-	allPlugins := []volume.VolumePlugin{}
 
-	// The list of plugins to probe is decided by this binary, not
-	// by dynamic linking or other "magic".  Plugins will be analyzed and
-	// initialized later.
-
-	// Each plugin can make use of VolumeConfig.  The single arg to this func contains *all* enumerated
-	// options meant to configure volume plugins.  From that single config, create an instance of volume.VolumeConfig
-	// for a specific plugin and pass that instance to the plugin's ProbeVolumePlugins(config) func.
-
-	// HostPath recycling is for testing and development purposes only!
-	hostPathConfig := volume.VolumeConfig{
-		RecyclerMinimumTimeout:   int(config.PersistentVolumeRecyclerConfiguration.MinimumTimeoutHostPath),
-		RecyclerTimeoutIncrement: int(config.PersistentVolumeRecyclerConfiguration.IncrementTimeoutHostPath),
-		RecyclerPodTemplate:      defaultScrubPod,
-		ProvisioningEnabled:      config.EnableHostPathProvisioning,
-	}
-	if err := kctrlmgr.AttemptToLoadRecycler(config.PersistentVolumeRecyclerConfiguration.PodTemplateFilePathHostPath, &hostPathConfig); err != nil {
-		return nil, fmt.Errorf("could not create hostpath recycler pod from file %s: %+v", config.PersistentVolumeRecyclerConfiguration.PodTemplateFilePathHostPath, err)
-	}
-	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins(hostPathConfig)...)
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
 
-	nfsConfig := volume.VolumeConfig{
-		RecyclerMinimumTimeout:   int(config.PersistentVolumeRecyclerConfiguration.MinimumTimeoutNFS),
-		RecyclerTimeoutIncrement: int(config.PersistentVolumeRecyclerConfiguration.IncrementTimeoutNFS),
-		RecyclerPodTemplate:      defaultScrubPod,
+// newPersistentVolumeRecyclerPodTemplate provides a function which makes our recycler pod template for use in the kube-controller-manager
+// this is a stop-gap until the kube-controller-manager take a pod manifest
+func newPersistentVolumeRecyclerPodTemplate(recyclerImageName string) func() *v1.Pod {
+	return func() *v1.Pod {
+		uid := int64(0)
+		defaultScrubPod := volume.NewPersistentVolumeRecyclerPodTemplate()
```

Comment:
wouldn't it recursively call `newPersistentVolumeRecyclerPodTemplate()` (the one that starts 3 lines above)?  IMO you need proper closure:

```go
func newPersistentVolumeRecyclerPodTemplate(...) {
    oldTemplateFunc := volume.NewPersistentVolumeRecyclerPodTemplate()
    return func() *v1.Pod {
		uid := int64(0)
		defaultScrubPod := oldTemplateFunc()
...
```

I am curious of test results...